packages/origin-device-registry-irec-local-api/src/device/device.entity.ts

```diff
@@ -0,0 +1,71 @@
+import { DeviceStatus } from '@energyweb/origin-backend-core';
+import { ExtendedBaseEntity } from '@energyweb/origin-backend-utils';
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({ name: 'irec_device_registry_device' })
+export class Device extends ExtendedBaseEntity {
+    constructor(device: Partial<Device>) {
+        super();
+        Object.assign(this, device);
+    }
+
+    @PrimaryGeneratedColumn('uuid')
+    id: string;
```

Comment:
Can `code` be used as the primary key, instead of generating a new id?
```
@PrimaryColumn('uuid')
code: string;
```

Reply:
Good observation, but in PGSQL the primary key has to be either guid or serial (auto increment), while code is the string.